src/js/components/Button.js

```diff
@@ -28,9 +28,9 @@ export default class Button extends Component {
     event.preventDefault();
 
     if ('push' === method) {
-      router.push(path);
+      (router.history || router).push(path.path || path);
```

Comment:
Button does not have the concept of `active`. So we don't need support for `path` as an object. it is just a regular string. So this line can be `(router.history || router).push(path)`